iXBRLViewerPlugin/viewer/src/i18n/en/taxonomies.json

```diff
@@ -0,0 +1,15 @@
+{
+    "http_//fasb_org/us-gaap/2022": "US GAAP 2022",
```

Comment:
I somewhat worry this file will become difficult to maintain, either we don't remember to update it or we end up adding many things to it.  I can already guess we're going to get support requests from customers (or our own professional services team) asking about inconsistencies.  I wonder if it would be nicer to allow the names to be passed on the command-line when generating a viewer so that the ixbrl-viewer.js doesn't need to be rebuilt to add new names? Long-term, it would be nice if we could figure out how taxonomies could just provide these terse labels themselves, perhaps with a generic linkbase and a namespace-as-resource?

@derekgengenbacher-wf Thoughts on how we might manage this?

Reply:
Not Derek :) but I like the idea of a "validation option" for it in tax-serv tbh, that seems completely reasonable.

Reply:
Another possible place to put it would be in the Taxonomy Package metadata.  That already provides the option of multilingual name and description.  It doesn't provide a short name option at the moment, but the schema is intentionally extensible, so it's something we could suggest.

It does already provide a separate version field.

The problem is how to then link it with specific namespaces, as a Taxonomy Package may define multiple concept-bearing namespaces.

This is probably something to raise with the Spec WG.

Reply:
To address the original point, I'm happy to move this to be a viewer-build-time input, rather than hard-coded into the JS, but I think it makes sense to provide some defaults.

I wonder if this is something we should actually address in Arelle?  For US taxonomies, Arelle already has the `edgartaxonomies.xml` file  which looks to provide exactly what we need, and is already being maintained in the Arelle codebase.

Reply:
> Another possible place to put it would be in the Taxonomy Package metadata.

I think we want it discoverable from the XBRL docs, and I wouldn't expect that metadata to be referenced by the DTS (sounds backwards).

> I'm happy to move this to be a viewer-build-time input, rather than hard-coded into the JS, but I think it makes sense to provide some defaults.

That seems reasonable to me.

> For US taxonomies, Arelle already has the edgartaxonomies.xml file

If adding ESEF files and exposing that with some kind of Python API seems reasonable to Herm, et al, then that seems somewhat more palatable than trying to maintain a registry in this repo.

Reply:
Raised https://gitlab.xbrl.org/base-spec/base-spec/-/issues/292 to discuss a standard XBRL mechanism for this.